devtools/onion.c

```diff
@@ -1,10 +1,16 @@
+#include <ccan/mem/mem.h>
```

Comment:
Erk, I just repurposed onion.c as a generic onion builder, which will clash horribly with this.

My approach with the TLV tests (which are also JSON) was to quote the spec in the test case (so if it's wrong or changes, check-source points it out) then pulls the JSON out of itself.  We should generalize that, IMHO.

Reply:
Yep, ran into quite a bit of trouble rebasing, but it's working fine now :-)